compaction/compaction.cc

```diff
@@ -156,17 +165,32 @@ static api::timestamp_type get_max_purgeable_timestamp(const table_state& table_
         if (compacting_set.contains(sst)) {
             continue;
         }
+        api::timestamp_type min_timestamp;
+        auto ts_stats = sst->get_ext_timestamp_stats();
+        if (!ts_stats.empty()) {
+            auto maybe_update_min_timestamp = [&] (sstables::ext_timestamp_stats_type t) {
```

Comment:
Good point.  In an earlier version it also checked that the timestamp does not equal max_timestamp (the default value), so the name fit better, but I removed this condition since it's wrong and we're interested in the default value alao in the absence of live data.

The absence of metadata in the sstable looks different, as disengaged ts_stats, or missing map entries (that can't happen today in this series, but the code is prepared for it anyhow)